### PR TITLE
Report error

### PR DESCRIPTION
**Pull request by dwulfse**
report error completed and modals tweaked.

### CODE REVIEW

File: components/Modal/Modal.jsx [hunk 1]
```diff
@@ -1,34 +1,32 @@
...
-                                    setModal(null);
-                                }}
-                                className="w-12 h-12 flex justify-center items-center"
-                            >
-                                <FontAwesomeIcon icon={faXmark} size={30} />
-                            </TouchableOpacity>
+            <TouchableWithoutFeedback onPress={() => setModal(null)}>
+                <View className="h-screen justify-center items-center bg-[#FFFFFF90]">
+                    <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
+                        <View className="h-1/2 w-10/12 border-2 rounded-xl bg-white">
```
Comment (dwulfse) [inline components/Modal/Modal.jsx]: h-1/2 shaaaaame

File: components/Modal/Modal.jsx [hunk 2]
```diff
@@ -1,34 +1,32 @@
...
-                            <TouchableOpacity
-                                onPress={() => {
-                                    setModal(null);
-                                }}
-                                className="w-12 h-12 flex justify-center items-center"
-                            >
-                                <FontAwesomeIcon icon={faXmark} size={30} />
-                            </TouchableOpacity>
+            <TouchableWithoutFeedback onPress={() => setModal(null)}>
+                <View className="h-screen justify-center items-center bg-[#FFFFFF90]">
```
Comment (DanWilkins2107) [inline components/Modal/Modal.jsx]: Can you not make this View a Touchable without feedback?

Comment (DanWilkins2107) [reply components/Modal/Modal.jsx]: and not wrap it with one?

File: components/Modal/Modal.jsx (hunk 1 above)
Comment (DanWilkins2107) [reply components/Modal/Modal.jsx]: Yeah I mean maybe we need to just pass in heights if we need a different height :( annoying tho

File: components/ProfileElements/HelpModal.jsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import { icon } from "@fortawesome/fontawesome-svg-core/import.macro";
+
+const furtherInfo = {
+    ReportError:
+        "Describe what happened in detail. If you can, include the steps you took to get to this point.",
+    AskQuestion: "Further details about your question.",
+    SuggestImprovement: "Further details about your suggestion.",
+};
+
+export default function HelpModal(props) {
```
Comment (DanWilkins2107) [inline components/ProfileElements/HelpModal.jsx]: Don't need to pass in props

File: components/Modal/Modal.jsx (hunk 1 above)
Comment (DanWilkins2107) [reply components/Modal/Modal.jsx]: Yeah looking at it we really need heights passed in if we can't use h-fit they all look really weird

File: components/ProfileElements/HelpModal.jsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        } catch (error) {
+            setAlert("Error sending report", "bg-red-500", icon({ name: "circle-exclamation" }));
+        }
+    };
+
+    return (
+        <View className="flex flex-col h-full items-center">
+            <Text className="text-2xl font-extrabold text-center mb-1">Help</Text>
+            <View className="flex-1 flex-col m-4">
+                <DropDownPicker
```
Comment (DanWilkins2107) [inline components/ProfileElements/HelpModal.jsx:58]: Don't know how easy this is, but it looks kinda weird with the different border widths when the dropdown is there?

File: components/UserDashboard/ProductionButton.jsx [hunk 5]
```diff
@@ -20,11 +20,11 @@ export default function ProductionButton({ navigation, production }) {
             <View className="w-max flex flex-row mt-2">
                 {Object.keys(production.participants)
                     .slice(0, 5)
-                    .map((id) => {
+                    .map((id, index) => {
                         return (
                             <>
                                 <ProfilePicture
-                                    key={id}
+                                    key={index}
```
Comment (DanWilkins2107) [inline components/UserDashboard/ProductionButton.jsx]: Don't think this has fixed it :/

**Review by DanWilkins2107 — CHANGES_REQUESTED** [review]
Good apart from small changes

File: components/Modal/Modal.jsx (hunk 1 above)
Comment (dwulfse) [reply components/Modal/Modal.jsx]: got it to work finally!! without needing to pass height

File: components/Modal/Modal.jsx (hunk 2 above)
Comment (dwulfse) [reply components/Modal/Modal.jsx]: done. had to use Pressable instead cos TouchableWithoutFeedback cant have multiple children and cant take a classname but defo much cleaner now

File: components/UserDashboard/ProductionButton.jsx (hunk 5 above)
Comment (dwulfse) [reply components/UserDashboard/ProductionButton.jsx]: seems like it has for me. when does the error come up? is it only for lots of participants?

Comment (dwulfse) [reply components/UserDashboard/ProductionButton.jsx]: actually surely the error is cos the view didn't have a key so I've given them keys now

**Review by DanWilkins2107 — APPROVED** [review]

**Pull request merged**